### PR TITLE
docs: fix site document title flash during page loading 

### LINKED ISSUE #44514: 访问网站时 document.title 会跳一下
https://github.com/ant-design/ant-design/assets/507615/47b875ad-b2c5-49ee-9d93-fd146b9b1b02

### PR DESCRIPTION
**Pull request by rickythink**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

close #44514

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   fix site document title flash during page loading         |
| 🇨🇳 Chinese |    修复网站加载时标题闪烁的问题       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at fde3f2c</samp>

Improved the handling of title and description for the index page of the documentation site. Used `.dumi/theme/layouts/DocLayout/index.tsx` to conditionally render the default or custom values.

### <samp>🤖 Generated by Copilot at fde3f2c</samp>

* Import `useState` hook from React to manage index page title and description state ([link](https://github.com/ant-design/ant-design/pull/44631/files?diff=unified&w=0#diff-bedc1b4864dae1cd42aff9682ee09c104a08db60b785929f6bede4898dc5fa0bL5-R5))
* Add helper functions `hasTitle` and `hasMeta` to check document title and meta tags ([link](https://github.com/ant-design/ant-design/pull/44631/files?diff=unified&w=0#diff-bedc1b4864dae1cd42aff9682ee09c104a08db60b785929f6bede4898dc5fa0bR30-R47))
* Add state variables `showDefaultTitle` and `showDefaultDesc` to control default title and description rendering ([link](https://github.com/ant-design/ant-design/pull/44631/files?diff=unified&w=0#diff-bedc1b4864dae1cd42aff9682ee09c104a08db60b785929f6bede4898dc5fa0bR57-R67))
* Use `useLayoutEffect` hook to update state variables based on prerendered title and meta tags ([link](https://github.com/ant-design/ant-design/pull/44631/files?diff=unified&w=0#diff-bedc1b4864dae1cd42aff9682ee09c104a08db60b785929f6bede4898dc5fa0bR76-R85))
* Conditionally render title and meta tags based on state variables ([link](https://github.com/ant-design/ant-design/pull/44631/files?diff=unified&w=0#diff-bedc1b4864dae1cd42aff9682ee09c104a08db60b785929f6bede4898dc5fa0bL101-R146))

### CODE REVIEW

File: .dumi/theme/layouts/DocLayout/index.tsx [hunk 1]
```diff
@@ -36,6 +54,17 @@ const DocLayout: React.FC = () => {
...
+  const isIndexPage = useMemo(
+    () =>
+      ['', '/'].some((path) => path === pathname) ||
+      ['/index'].some((path) => pathname.startsWith(path)),
+    [pathname],
+  );
+
+  // gen default title and description for index page, other pages should have their own title and description by definitions in markdown files
+  const [showDefaultTitle, setShowDefaultTitle] = useState(!!isIndexPage);
+  const [showDefaultDesc, setShowDefaultDesc] = useState(!!isIndexPage);
```
Comment (afc163) [inline .dumi/theme/layouts/DocLayout/index.tsx:66]: 是啥原因导致的 title 丢了，现在这个修复方式感觉有点脏。

Comment (rickythink) [reply .dumi/theme/layouts/DocLayout/index.tsx]: You raise a good point. My prior proposal did not account for all factors and thus was not the best approach.

Looking closer, it seems the root cause is that `DocLayout` introduces a shared `Helmet` component with **title** and **description** settings, but `SidebarLayout` also imports `CommonHelmet` which configures **title** and **description**. This _duplication_ can lead to title flickering at different render times.

The solution is fairly straightforward - remove the title configuration in `DocLayout`, provide a dedicated title just for the homepage, and leave other layouts to leverage the existing CommonHelmet. This ensures each page has its own singular title setting.

**Comment by afc163** [comment]
It would cause a suspense fallback during page load.


https://github.com/ant-design/ant-design/assets/507615/174818cf-cf49-4e81-9b1c-f0eb8182d5d6

**Comment by rickythink** [comment]
> It would cause a suspense fallback during page load.
> 
>  2023-09-07.3.53.08.mov

I appreciate you bringing this issue to my attention. 
It is a complex matter that I do not yet fully understand. I will need some time to carefully consider the details in order to come up with an appropriate solution.

**Review by afc163 — APPROVED** [review]

**Pull request merged**